core/arch/arm/mm/core_mmu.c

```diff
@@ -959,6 +959,10 @@ static void check_pa_matches_va(void *va, paddr_t pa)
 		return;
 	}
 #ifdef CFG_WITH_PAGER
+	if (v >= CFG_TEE_LOAD_ADDR && v < core_mmu_linear_map_end) {
+		TEE_ASSERT(v == pa);
```

Comment:
@jenswi-linaro +1 for fixing our `assert()` wrt. `NDEBUG` to properly follow the standard.

I also agree that `TEE_ASSERT` is confusing (is the expression always evaluated?), and I think your suggestion to simply remove it is good. While we're at it, I would suggest adding a reason string, like the Linux panic function: `if (!expr) panic("Error message")`. The implementation could print the message only, or the message plus the file/line info, depending on the debug level.